### ISSUE TITLE
Exception on newest version of elasticsearch

### ISSUE THREAD

**Issue by odmarkj** [jeskew/amazon-es-php#5]
v5.3.2 causes a "Host must be a string" to be thrown.

```
$provider = CredentialProvider::fromCredentials(
    new Credentials($_SERVER['AWS_ACCESS_KEY_ID'], $_SERVER['AWS_SECRET_KEY'])
);

$handler = new ElasticsearchPhpHandler($_SERVER['AWS_REGION'], $provider);

$client = ClientBuilder::create()
                ->setHandler($handler)
                ->setHosts(['https://url.com:443'])
                ->build();

$results = $client->search($params);
```

Reverting to v5.3.0 fixes the issue.

This just started happening a few days ago.

I may eventually submit a PR for a fix, but don't have the time right now.

**Comment by farnsworthMars** [comment]
I saw the same thing - added this to my composer.json require block to lock elasticsearch/elasticsearch to 5.3.0 and the problem went away.  
```"elasticsearch/elasticsearch": "5.3.0",```

**Comment by jeskew** [comment]
This looks like it was caused by [this change](https://github.com/elastic/elasticsearch-php/commit/1621094).

**Comment by rafaeltovar** [comment]
The same for me. With `elasticsearch/elasticsearch` `5.3.0` works correctly, but not with `5.3.1`.

**Comment by rafaeltovar** [comment]
I did a pull request with fix. 
https://github.com/jeskew/amazon-es-php/pull/6

**Comment by jeskew** [comment]
Fixed in #6
